Libs/Widgets/ctkWidgetsUtils.cpp

```diff
@@ -113,3 +115,28 @@ QImage ctk::kwIconToQImage(const unsigned char *data, int width, int height, int
     }
   return image.copy();
 }
+
+//----------------------------------------------------------------------------
+QIcon ctk::getColorizedIcon(const QString& hexColor, const QString& resourcePath)
+{
+  QIcon icon;
+  if (hexColor == "#000000")
+  {
+    // resource icons are already colored black by default
+    icon = QIcon(resourcePath);
+  }
+  else
+  {
+    QFile file(resourcePath);
+    file.open(QIODevice::ReadOnly);
+    QByteArray data = file.readAll();
+    QString newHexString = "fill=\"" + hexColor;
+    QByteArray newByteArray = newHexString.toLocal8Bit();
+    const char *newHexChar = newByteArray.data();
+    data.replace("fill=\"#000000", newHexChar);
+    QPixmap pixmap;
+    pixmap.loadFromData(data);
+    icon = QIcon(pixmap);
```

Comment:
A QIcon can hold several pixmaps, so that it can retrieve the one most suitable resolution (or generate pixmaps using an engine), depending on the required size and scaling.

This proposed method would only add a single pixmap, which could result in blurry icons on hi-dpi screens.

The difficulty is that Qt does not offer an API to create a QIcon from an XML string, only from a file or icon engine.

There are icon engine based solutions (https://smtk.readthedocs.io/en/latest/doc/reference/smtk/html/classsmtk_1_1extension_1_1SVGIconEngine.xhtml#details), but it feels like an overkill. Maybe if we add dynamic recoloring there (see for example https://github.com/rizinorg/cutter/blob/dev/src/common/SvgIconEngine.cpp) then that could somewhat justify the complexity.

We could create pixmaps at the original, 2x, and 3x resolutions then store them all in the QIcon object. However, this would be probably slower (we would render 3 times because we would not know the required sizes in advance) and it would still be a somewhat limited (not as dynamic as generating pixmaps at any resolution with an icon engine).

Probably the simplest solution would be to write back the modified XML string to a file, instantiate the QIcon from it, then delete the file.

Reply:
I've tested this and I can confirm that unfortunately on hi-dpi screens the icons appear blocky:

![image](https://user-images.githubusercontent.com/307929/192557252-3e59e203-d853-46e3-9910-db7b74af7c7d.png)